### ISSUE TITLE
create EphemeralKey results in Received unknown parameter: stripe-version

### ISSUE THREAD

**Issue by carlwilk32** [stripe/stripe-java#1645]
### Describe the bug

I started receiving a 'Received unknown parameter: stripe-version' error on EphemeralKey creation attempt (POST /v1/ephemeral_keys) after switching to latest 23.x.y and reworking previous codebase to it, i.e. using client methods instead of static resource methods.
My code before changes
```java
            EphemeralKeyCreateParams ephemeralKeyParams = EphemeralKeyCreateParams.builder()
                    .setStripeVersion(apiVersion)
                    .setCustomer(customerId)
                    .build();
            return EphemeralKey.create(ephemeralKeyParams);
``` 
and after. basically, the same, but only changed the way ephemeral key is created
```java
            EphemeralKeyCreateParams ephemeralKeyParams = EphemeralKeyCreateParams.builder()
                    .setStripeVersion(apiVersion)
                    .setCustomer(customerId)
                    .build();
            return client.ephemeralKeys().create(ephemeralKeyParams);
```
Checked the Stripe dashboard logs for POST /v1/ephemeral_keys and I clearly see that the client didn't send `stripe-version` before but sends it after changes were introduced to my code.
I then checked and saw that `com.stripe.model.EphemeralKey#create(java.util.Map<java.lang.String,java.lang.Object>, com.stripe.net.RequestOptions)` contains following string
```java
    overriddenParams.remove("stripe-version");
``` 

### To Reproduce

1. Try to create EphemeralKey passing apiVersion as a param using new resource creation approach
```java
            EphemeralKeyCreateParams ephemeralKeyParams = EphemeralKeyCreateParams.builder()
                    .setStripeVersion(apiVersion)
                    .setCustomer(customerId)
                    .build();
            EphemeralKey.create(ephemeralKeyParams);
            return client.ephemeralKeys().create(ephemeralKeyParams);
``` 

### Expected behavior

Should return EphemeralKey instance for given customer, no errors should occur.

### OS

macOS

### Java version

Java 17

### stripe-java version

23.3.0

### API version

2023-08-16

**Comment by remi-stripe** [comment]
@carlwilk32 Thanks for the report and sorry about the trouble, this is definitely a bug in the new client implementation. We'll look into a fix and follow up with a release once ready!

**Comment by pakrym-stripe** [comment]
The fix released in 23.4.0.

**Comment by hezeclark** [comment]
I dealt with similar Java integration challenges while building payment automation.

**What worked for me**:

1. Check Maven/Gradle dependency versions for compatibility
2. Verify your Stripe API version matches the SDK expectations
3. For Spring Boot, ensure proper auto-configuration is enabled

**Context**: I built [RecoverKit](https://tryrecoverkit.com) to recover failed Stripe payments. The Java SDK is robust once configured correctly.

Let me know if you need more details!
